Traversable homework (Ilia Gusevskii)

IamAfli opened the pull request.

IamAfli committed:
Feat: mapping() in Mapping released

IamAfli committed:
Refactor: methods addOneYear and changeAllQAToUppercase now wrok, using MapHelper

IamAfli committed:
Feat: lazy_mapping() and LazyMapHelper released

Commit:
Feat: LazyFlatMapHelper released

Commit:
initial commit

Commit:
Methods from and toList released

Commit:
Feat: method filter released

Commit:
Feat: mathod map released

Commit:
flatMap method released

senia-psm commented on src/test/java/lambda/part3/exercise/Mapping.java:
```diff
@@ -224,4 +199,59 @@ public void lazy_mapping() {
+        }
+
+        public LazyFlatMapHelper<T, R> filter (Predicate<T> predicate){
+            return new LazyFlatMapHelper<>(
```
You could call `flatMap` here.

senia-psm commented on src/test/java/lambda/part3/exercise/Mapping.java:
```diff
@@ -224,4 +199,59 @@ public void lazy_mapping() {
+            return result;
+        }
+
+        public LazyFlatMapHelper<T, R> filter (Predicate<T> predicate){
```
It should be `Predicate<R>`

IamAfli commented:
This pull request (#128) is only about Traversable<T> task, it is
implemented in class TraversableTests:
https://github.com/IamAfli/lambda/blob/traversable/src/test/java/lambda/part3/exercise/TraversableTests.java

And LazyFlatMapHelper<T, R> is released is pull request #113.
However, i do not understand your comment about Predicate<R> above, because
i did the same way,when i implemented method filter in LazyFlatMapHelper.

I can write you in slack, if it is more comfortable.


2017-06-23 22:16 GMT+03:00 Simon Popugaev <notifications@github.com>:

> *@senia-psm* commented on this pull request.
> ------------------------------
>
> In src/test/java/lambda/part3/exercise/Mapping.java
> <https://github.com/java8-course/lambda/pull/128#discussion_r123824901>:
>
> > +        }
> +
> +        public static <T> LazyFlatMapHelper<T, T> from(List<T> list) {
> +            return new LazyFlatMapHelper<>(list, Arrays::asList);
> +        }
> +
> +        public List<R> force() {
> +            List<R> result = new ArrayList<>();
> +            for (T element : list) {
> +                result.addAll(function.apply(element));
> +            }
> +            return result;
> +        }
> +
> +        public LazyFlatMapHelper<T, R> filter (Predicate<T> predicate){
> +            return new LazyFlatMapHelper<>(
>
> You could call flatMap here.
> ------------------------------
>
> In src/test/java/lambda/part3/exercise/Mapping.java
> <https://github.com/java8-course/lambda/pull/128#discussion_r123824986>:
>
> > +            this.function = function;
> +        }
> +
> +        public static <T> LazyFlatMapHelper<T, T> from(List<T> list) {
> +            return new LazyFlatMapHelper<>(list, Arrays::asList);
> +        }
> +
> +        public List<R> force() {
> +            List<R> result = new ArrayList<>();
> +            for (T element : list) {
> +                result.addAll(function.apply(element));
> +            }
> +            return result;
> +        }
> +
> +        public LazyFlatMapHelper<T, R> filter (Predicate<T> predicate){
>
> It should be Predicate<R>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/java8-course/lambda/pull/128#pullrequestreview-46068899>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AUMr33J9XNfCtS-cREMmAI-4xdAOcSZhks5sHA74gaJpZM4OD5-0>
> .
>



-- 
С уважением, Гусевский Илья

senia-psm commented:
Remove unrelated changes from this PR.

> i did the same way

This only means that you have the same error in 2 `PR`s.

Add a test with `LazyFlatMapHelper.from(listOfStrings).map(s -> s.length()).filter(l -> l > 0)`.
